src/board/processor.ts

```diff
@@ -102,16 +105,18 @@ export class Processor extends EventEmitter<ProcessorEvents> {
 
   private cycle() {
     const pc: Word = this.registers.readRegister(Register.PC)
-    const opcode: Halfword = this.memory.readHalfword(pc)
-    if (opcode.value === END_OF_CODE.value) {
+    const opcode = [this.memory.readHalfword(pc)]
+    if (opcode[0].value === END_OF_CODE.value) {
       this.halt()
       this.emit('endOfCode')
       return
     }
-    this.instructions
-      .getExecutor(opcode)
-      .executeInstruction(opcode, this.registers, this.memory)
-    this.registers.writeRegister(Register.PC, pc.add(2))
+    const executor = this.instructions.getExecutor(opcode[0])
+    for (let i = 1; i < executor.opcodeLength; i++) {
+      opcode.push(this.memory.readHalfword(pc.add(i)))
```

Comment:
Little bug. PC register should be incremented by `i * 2` to read the correct halfword from the memory.
Fixed this in the branch vcb-13-unconditional-jump.